pkg/util/intlresource/intlresource.go

```diff
@@ -0,0 +1,47 @@
+package intlresource
+
+import (
+	"github.com/authgear/authgear-server/pkg/util/intl"
+	"github.com/authgear/authgear-server/pkg/util/resource"
+)
+
+func Prepare(
+	resources []resource.ResourceFile,
+	view resource.EffectiveResourceView,
+	extractLanguageTag func(resrc resource.ResourceFile) string,
+	add func(langTag string, resrc resource.ResourceFile) error,
+) error {
+	supportedLanguageTags := view.SupportedLanguageTags()
+
+	supportedSet := make(map[string]struct{})
+	for _, tag := range supportedLanguageTags {
+		supportedSet[tag] = struct{}{}
+	}
+
+	var fallbackResrc resource.ResourceFile
+	added := false
+	for _, resrc := range resources {
+		langTag := extractLanguageTag(resrc)
+		if langTag == intl.DefaultLanguage {
+			fallbackResrc = resrc
+		}
+		// Ignore resources in unsupported languages.
+		_, supported := supportedSet[langTag]
+		if !supported {
+			continue
+		}
+		err := add(langTag, resrc)
+		if err != nil {
+			return err
+		}
+		added = true
```

Comment:
Would like to confirm the behavior of the `Prepare` function
1. is to ensure `intl.DefaultLanguage` is added even it is not listed in `supportedLanguageTags`?
OR
2. is to ensure there is at least one language added ? 

Now the function is doing 2.

Actucally I found a problem in `translationJSON` Descriptor that. If developer removed `en` language in supported language, e.g. the app only support `zh-TW`. If there is new items which is missing in `zh-TW/translation.json`, the app will crash due to missing translation item. Not sure always adding `en` can help solve the problem? Since we defined the `en`.

But if user provide `en/translation.json` that cannot have all items, will the same problem occur again?

The problem seems a bit complicated, see if we need to discuss for that?

Reply:
> But if user provide en/translation.json that cannot have all items, will the same problem occur again?

Double check and found, missing items in user defined `en/translation.json` is not a problem.